File: plugins/flutter_bg_location_plugin/android/src/main/kotlin/com/ezride/flutter_bg_location_plugin/handlers/LocationServiceStopHandler.kt
package com.ezride.flutter_bg_location_plugin.handlers

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import android.content.Context
import android.content.Intent
import android.util.Log
import com.ezride.flutter_bg_location_plugin.services.LocationService

class LocationServiceStopHandler : Handler{

    override val callMethod : String = CallMethods.LOCATION_SERVICE_STOP;

    override fun handler(context: Context, call: MethodCall, result: MethodChannel.Result){
        Log.d("FlutterLocationPlugin", "stopTracking invoked");
        val isStoped =  LocationService.stopTracking(context);
        result.success(isStoped);
    }
}

File: plugins/flutter_bg_location_plugin/ios/Classes/Handler/LocationServiceStopHandler.swift
import  Flutter

final class LocationServiceStopHandler: Handler {

    let callMethod = CallMethods.LOCATION_SERVICE_STOP

    func handler(call: FlutterMethodCall, result: @escaping FlutterResult) {
        print("FlutterLocationPlugin", "stopTracking invoked")

        let isStoped =  PluginContext.shared.locationService.stopTracking()
        result(isStoped)
    }
}